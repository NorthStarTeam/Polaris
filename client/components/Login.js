import React from 'react';

const Login = () => {
  return (
    <div>
      <h1>We Login to Profile</h1>
      
      <div className='login-box'>
        <form className='login-form'>

          {/* Username Input */}
          <input className='login-input' placeholder='username' type='text' ></input>

          {/* Password Input */}
          <input className='login-input' placeholder='password' type='text'></input>

          {/* Submit Login Request */}
          <input className='login-button' type='submit' value='Login'></input>
        </form>
      </div>

    </div>
  );
};

export default Login;
