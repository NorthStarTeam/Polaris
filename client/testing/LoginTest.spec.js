import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Login from '../components/Login';


describe('Login', () => {

  //snapshot test for login page, takes a picture of how the html should look
  //and compares differences, can be updated by just pressing u in the test suite
  //when a snapshot is failed - useful for testing UI doesn't change unexpectedly
  test('snapshot renders', () => {
    const component = renderer.create(<Login />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  //enzyme test for login page, 
  it('renders the inner div with class login-box ', () => {
    const wrapper = mount(<Login />)
    expect(wrapper.find('div.login-box').length).toEqual(1)
  })

  it(`renders the form within the inner div above`, () => {
    const wrapper = mount(<Login />)
    expect(wrapper.find('form.login-form').length).toEqual(1)
  })

  // it('send form data correctly on click', ()=>{

  //   const wrapper = mount(<Login />)
  //   wrapper.find('input.login-input').at(0).simulate('click')
    
  // })
});

