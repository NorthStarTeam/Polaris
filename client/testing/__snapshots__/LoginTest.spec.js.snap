// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot renders 1`] = `
<div>
  <h1>
    Log In
  </h1>
  <div
    className="login-box"
  >
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <input
        className="login-input"
        onChange={[Function]}
        placeholder="Username"
        type="text"
      />
      <input
        className="login-input"
        onChange={[Function]}
        placeholder="Password"
        type="text"
      />
      <input
        className="login-button"
        type="submit"
        value="Login"
      />
    </form>
  </div>
</div>
`;
